bundles/org.openhab.binding.linktap/src/main/java/org/openhab/binding/linktap/internal/Firmware.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.linktap.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link Firmware} class defines the firmware version.
+ *
+ * @author David Goodyear - Initial contribution
+ */
+@NonNullByDefault
+public class Firmware {
+    String raw;
+    int buildVer;
+    int hwVer;
+
+    public Firmware(final @Nullable String fwVersion) {
+        raw = "S00000000";
+        hwVer = 0;
+        buildVer = 0;
+
+        if (fwVersion == null || fwVersion.length() < 7) {
+            return;
+        } else {
+            raw = fwVersion;
+            buildVer = Integer.parseInt(raw.substring(1, 7));
```

Comment:
A theoreticall `IndexOutOfBoundsException` can occure:
The substring is zero based. Previously there was a length check for `< 7` and that will garantuee that index 6 is avilable, but it does not say index 7 or more is available. 

If i'm not mistake the check should be `<= 7`

Reply:
I think its correct, these index's always get me, so had to double check quickly with raw code.

private String test6 = "012345";
private String test7 = "0123456";
private String test8 = "01234567";
    
    public String runTest6() {
        if (test6.length() < 7) {
            return "NA";
        }
        return test6.substring(1,7);
    }

    public String runTest7() {
        if (test7.length() < 7) {
           return "NA";    
        }
        return test7.substring(1,7);
       
    }
    
    public String runTest8() {
        if (test8.length() < 7) {
             return "NA";
        }
        return test8.substring(1,7);
    }

test6 gives NA as expected. test7 and test8 give 1,2,3,4,5,6 as expected, so I thinks ok, and replicates the current code flow without the extra null check. I know im missed these before, so had to be sure :)

Reply:
Thanks, I have to sleep. My logic was wrong.